application/forms/IcingaDependencyForm.php

```diff
@@ -212,18 +213,50 @@ protected function addObjectsElement()
 
         if (!empty($sentParent) || $dependency->isApplyRule()) {
             $parentService = $dependency->get('parent_service');
+            if ($parentService === null) {
+                $parentServiceVar = $dependency->get('parent_service_by_name');
+                if ($parentServiceVar) {
+                    $parentService = '$' . $parentServiceVar . '$';
+                }
+            }
+
+            $applyTo = $this->getSentOrObjectValue('apply_to');
+
+            $description = $this->translate(
+                'Optional. The parent service. If omitted this dependency'
+                . ' object is treated as host dependency.'
+            );
+
+            if ($applyTo === 'service') {
+                $description .= $this->translate(
+                    ' You might want to refer to Service Custom Variables via $service.vars.varname$'
+                );
+            }
+
+            $callbackValidator = new Zend_Validate_Callback(function ($value) {
+                $applyTo = $this->getSentValue('apply_to');
+                if ($applyTo === 'service') {
+                    return true;
+                }
+
+                return ! $this->isCustomVar($value);
```

Comment:
This should complain about service custom variables in case of a host dependency. Such shouldn't be silently accepted and rendered as strings.

Reply:
this condition changed to `$applyTo === 'service' && $this->isCustomVar($value)` as we need to check whether the parent service name is a custom variable only for service dependencies.